pootle/apps/pootle_app/management/commands/create_project.py

```
@@ -0,0 +1,149 @@
+        )
+        parser.add_argument(
+            "--checkstyle",
+            action="store",
+            dest="checkstyle",
+            choices=PROJECT_CHECKERS.keys(),
+            default="standard",
+            help="Quality check styles. Example: {0}"
+                 .format(" | ".join(PROJECT_CHECKERS.keys())),
+        )
+
+    def handle(self, **options):
+        self.create_project(**options)
+
+    def create_project(self, **options):
+        """Immitates the behaviour of the admin interface
+            for creating a project from command line."""
+
+        try:
+            source_language_obj = Language.objects.get(code=options["sourcelang"])
+        except Language.DoesNotExist:
+            raise CommandError("Source language %s does not exist"
+                               % options["sourcelang"])
+
+        if(options["contact"]):
+            try:
+                validate_email(options["contact"])
+            except ValidationError as e:
+                raise CommandError(e)
+
+        new_project = Project.objects.create(
+            code=options["code"],
+            fullname=options["name"] or options["code"].capitalize(),
+            checkstyle=options["checkstyle"],
+            source_language=source_language_obj,
+            filetypes=options["filetypes"] or ["po"],
+            report_email=options["contact"] or "",
+            disabled=options["disabled"])
+
+        if(options["preset_mapping"]):
+            new_project.config["pootle_fs.translation_mapping"] = dict(
+                default=FS_PRESETS[options["preset_mapping"]][0])
+        else:
+            new_project.config["pootle_fs.translation_mapping"] = dict(
+                default=options["mapping"])
+
+        if(options["fs_type"]):
+            new_project.config["pootle_fs.fs_type"] = options["fs_type"]
+
+        if(options["fs_url"]):
+            new_project.config["pootle_fs.fs_url"] = options["fs_url"]
```

Is this not checked before assignment?

---

it should raise an error if the assignment is wrong - i need to follow up

---

When it comes to a command I would expect all provided input is sanitized before doing anything. With this pattern you might end creating a project with partial or wrong config.

---

hmm, lets make this command atomic, and then it wont - if it triggered an error i would have caught it